scipy/spatial/transform/rotation.py

```diff
@@ -0,0 +1,868 @@
+            return self._quat.copy()
+
+    def as_dcm(self):
+        """Represent rotations as direction cosine matrices.
+
+        3D rotations can be represented using direction cosine matrices, which
+        are 3 x 3 real orthogonal matrices with determinant equal to +1 [1]_.
+
+        Returns
+        -------
+        dcm : `numpy.ndarray`, shape (3, 3) or (N, 3, 3)
+            Shape depends on shape of inputs used for initialization.
+
+        References
+        ----------
+        .. [1] `Direction Cosine Matrix
+                <https://en.wikipedia.org/wiki/Rotation_matrix#In_three_dimensions>`_
+        """
+
+        x = self._quat[:, 0]
+        y = self._quat[:, 1]
+        z = self._quat[:, 2]
+        w = self._quat[:, 3]
+
+        x2 = x * x
+        y2 = y * y
+        z2 = z * z
+        w2 = w * w
+
+        xy = x * y
+        zw = z * w
+        xz = x * z
+        yw = y * w
+        yz = y * z
+        xw = x * w
+
+        num_rotations = len(self)
+        dcm = np.empty((num_rotations, 3, 3))
+
+        dcm[:, 0, 0] = x2 - y2 - z2 + w2
+        dcm[:, 1, 0] = 2 * (xy + zw)
+        dcm[:, 2, 0] = 2 * (xz - yw)
+
+        dcm[:, 0, 1] = 2 * (xy - zw)
+        dcm[:, 1, 1] = - x2 + y2 - z2 + w2
+        dcm[:, 2, 1] = 2 * (yz + xw)
+
+        dcm[:, 0, 2] = 2 * (xz + yw)
+        dcm[:, 1, 2] = 2 * (yz - xw)
+        dcm[:, 2, 2] = - x2 - y2 + z2 + w2
+
+        if self._single:
+            return dcm[0]
+        else:
+            return dcm
+
+    def as_rotvec(self):
+        """Represent rotations as rotation vectors.
+
+        A rotation vector is a 3 dimensional vector which is co-directional to
+        the axis of rotation and whose norm gives the angle of rotation (in
+        radians) [1]_.
+
+        Returns
+        -------
+        rotvec : `numpy.ndarray`, shape (3,) or (N, 3)
+            Shape depends on shape of inputs used for initialization.
+
+        References
+        ----------
+        .. [1] `Rotation Vectors
+                <https://en.wikipedia.org/wiki/Axis%E2%80%93angle_representation#Rotation_vector>`_
+        """
+        quat = self._quat.copy()
+        # w > 0 to ensure 0 <= angle <= pi
+        quat[quat[:, 3] < 0] *= -1
+
+        angle = 2 * np.arctan2(np.linalg.norm(quat[:, :3], axis=1), quat[:, 3])
+
+        small_angle = (angle <= 1e-3)
+        large_angle = ~small_angle
+
+        num_rotations = len(self)
+        scale = np.empty(num_rotations)
+        scale[small_angle] = (2 + angle[small_angle] ** 2 / 12 +
+                              7 * angle[small_angle] ** 4 / 2880)
+        scale[large_angle] = (angle[large_angle] /
+                              np.sin(angle[large_angle] / 2))
+
+        rotvec = scale[:, None] * quat[:, :3]
+
+        if self._single:
+            return rotvec[0]
+        else:
+            return rotvec
+
+    def as_euler(self, seq, degrees=False):
+        """Compute Euler angles for rotations with specified axis sequence.
```

Comment:
Maybe follow the format you used in other `as_` methods, i.e. "Represent [rotations, Rotation] as Euler angles."